tests/test_compound_comparison_levels.py

```diff
@@ -0,0 +1,218 @@
+                    {
+                        "sql_condition": (
+                            f"{col_is_match('first_name')} AND "
+                            f"{col_is_match('middle_name')} AND "
+                            f"{col_is_match('surname')}"
+                        ),
+                        "label_for_charts": "All three match",
+                    },
+                    {
+                        "sql_condition": sql_and_clauses_joined_with_ors,
+                        "label_for_charts": "2 out of 3 columns match",
+                    },
+                    cll.exact_match_level("first_name"),
+                    cll.exact_match_level("middle_name"),
+                    cll.exact_match_level("surname"),
+                    cll.else_level(),
+                ],
+            },
+        ],
+    }
+
+    linker = DuckDBLinker(df, settings)
+    all_cols_match_level = linker._settings_obj.comparisons[1].comparison_levels[1]
+    assert all_cols_match_level._is_exact_match
+    assert set(all_cols_match_level._exact_match_colnames) == {
+        "first_name",
+        "middle_name",
+        "surname",
+    }
+
+    linker.estimate_parameters_using_expectation_maximisation("l.city = r.city")
+
+
+def test_complex_compound_comparison_level():
+
+    # non-realistic example
+    df = pd.DataFrame(
+        [
+            {
+                "unique_id": 1,
+                "col_1": "a",
+                "col_2": "b",
+                "col_3": "c",
+                "col_4": "d",
+                "col_5": "e",
+                "col_6": "f",
+                "col_7": "g",
+            },
+            {
+                "unique_id": 2,
+                "col_1": "aa",
+                "col_2": "b",
+                "col_3": "cc",
+                "col_4": "d",
+                "col_5": "ee",
+                "col_6": "f",
+                "col_7": "gg",
+            },
+            {
+                "unique_id": 3,
+                "col_1": "a",
+                "col_2": "bb",
+                "col_3": "c",
+                "col_4": "dd",
+                "col_5": "e",
+                "col_6": "ff",
+                "col_7": "g",
+            },
+            {
+                "unique_id": 4,
+                "col_1": "aa",
+                "col_2": "bb",
+                "col_3": "cc",
+                "col_4": "d",
+                "col_5": "ee",
+                "col_6": "ff",
+                "col_7": "gg",
+            },
+        ]
+    )
+    A, B, C, D, E, F, G = (
+        "col_1_l = col_1_r",
+        "col_2_l = col_2_r",
+        "col_3_l = col_3_r",
+        "col_4_l = col_4_r",
+        "col_5_l = col_5_r",
+        "col_6_l = col_6_r",
+        "col_7_l = col_7_r",
+    )
+
+    complex_condition_sql = " OR ".join(
+        [
+            f"({A} AND {B})",
+            f"NOT ({C} AND {D})",
+            f"({B} AND NOT {E})",
+            f"({F} AND (NOT {A} AND {G}))",
```

Comment:
You could also check the new sqlglot `normlize` function is doing its job as expected here.

`A and B OR C` -> `(a OR c) AND (b OR c)`